### ISSUE TITLE
Potential Runtime Error in 'Conv' Layer Prefill with Long Sequences and Caching Enable

### ISSUE THREAD

**Issue by kx-h** [test-time-training/ttt-lm-pytorch#36]
I'm currently reading through the code and haven't run it yet, so my understanding might be inaccurate. I have a question about the Conv layer implementation in ttt.py [Conv](https://github.com/test-time-training/ttt-lm-pytorch/blob/main/ttt.py#L359) regarding the logic for the prefill stage when caching is **enabled.**

### Scenario

1. If we run the model without `causal_conv1d_fn`
2. The model is called for the first time with a long input sequence (prefill stage), where `seq_len > config.conv_kernel.`
3. `use_cache=True` is set, so a `cache_params` object is created and passed down. At this point, `cache_params.seqlen_offset` is `0`.

### Code Path:

Inside the `Conv` module's `__call__` method, my reading suggests the execution follows this path([line381)](https://github.com/test-time-training/ttt-lm-pytorch/blob/main/ttt.py#L381):

```python
    if causal_conv1d_fn is None:
       if cache_params is not None: -> True
          if cache_params.seqlen_offset > 0: -> False
```
This leads to the else block for first-time calls [(line392):](https://github.com/test-time-training/ttt-lm-pytorch/blob/main/ttt.py#L392)
```python
    else: 
        conv_state = nn.functional.pad(
            hidden_states,
            (self.config.conv_kernel - hidden_states.shape[-1], 0), )
        cache_params.conv_states_dic["pre_conv"][self.layer_idx].copy_(conv_state) 
        hidden_states = self.conv(hidden_states)[..., :seq_len]
```

### The potential Issue (if I do not misunderstand the code)

In this block, the nn.functional.pad function is called with a padding amount of `self.config.conv_kernel - hidden_states.shape[-1]`.

If the initial sequence length `hidden_states.shape[-1]` is greater than `self.config.conv_kernel` (which is common for prefilling), this value will be negative, which would cause a runtime error since `F.pad` does not accept negative padding.

Could you please clarify the intended execution flow here? As I'm only reading the code, it's very likely I've misunderstood something. 

Thank you for your time and for your great work on this project!
